Gems/ROS2/Code/Source/RobotImporter/SDFormat/ROS2SensorHooks.cpp

```diff
@@ -98,4 +107,181 @@ namespace ROS2::SDFormat
         return importerHook;
     }
 
+    SensorImporterHook ROS2SensorHooks::ROS2GNSSSensor()
+    {
+        SensorImporterHook importerHook;
+        importerHook.m_sensorTypes = AZStd::unordered_set<sdf::SensorType>{ sdf::SensorType::NAVSAT };
+        importerHook.m_supportedSensorParams = AZStd::unordered_set<AZStd::string>{ ">update_rate" };
+        importerHook.m_pluginNames = AZStd::unordered_set<AZStd::string>{ "libgazebo_ros_gps_sensor.so" };
+        importerHook.m_supportedPluginParams = AZStd::unordered_set<AZStd::string>{};
+        importerHook.m_sdfSensorToComponentCallback = [](AZ::Entity& entity,
+                                                         const sdf::Sensor& sdfSensor) -> SensorImporterHook::ConvertSensorOutcome
+        {
+            if (!sdfSensor.NavSatSensor())
+            {
+                return AZ::Failure(AZStd::string("Failed to read parsed SDFormat data of %s NavSat sensor", sdfSensor.Name().c_str()));
+            }
+
+            SensorConfiguration sensorConfiguration;
+            sensorConfiguration.m_frequency = sdfSensor.UpdateRate();
+            const AZStd::string messageType = "sensor_msgs::msg::NavSatFix";
+            Internal::AddTopicConfiguration(sensorConfiguration, "gnss", messageType, messageType);
+
+            if (entity.CreateComponent<ROS2GNSSSensorComponent>(sensorConfiguration, GNSSSensorConfiguration()))
+            {
+                return AZ::Success();
+            }
+            else
+            {
+                return AZ::Failure(AZStd::string("Failed to create ROS2 GNSS Sensor component"));
+            }
+        };
+
+        return importerHook;
+    }
+
+    SensorImporterHook ROS2SensorHooks::ROS2ImuSensor()
```

Comment:
This could be done, I have no preferences. There will be one more hook for _ROS2ContactSensorComponent_ and possibly some more implementation to map plugins' parameters.

Reply:
I am thinking of the future maintenance of this this file to it prevent from growing to thousands of lines, if we add many more sensors(such a Thermal, GPS, etc...).

Reply:
@jhanca-robotecai please split into separate files.

Reply:
Code chunks specific to different sensors were moved to separate _cpp_ files. I did not create separate headers for each one. The base _cpp_ file was kept with a shared method - more methods will be added soon.